### ISSUE TITLE
Floating element not triggered on key presses when used with useButton from @react-aria/button

### ISSUE THREAD

**Issue by nygardk** [floating-ui/floating-ui#2646]
Hello!

I have encountered the following issue when using `@floating-ui/react` together with [`@react-aria/button`](https://react-spectrum.adobe.com/react-aria/useButton.html):
Keyboard presses `Space` and `Enter` do not trigger the floating element as expected when the button is in focus. Mouse clicks do work as expected.

To test this issue, I created a [sandbox](https://codesandbox.io/p/sandbox/floating-ui-react-aria-usebutton-test-x8wsdy). Use `Tab` to focus on the buttons and test how pressing `Space` works.

I am not sure if this actually an issue of `floating-ui`, `react-aria` or a mistake in my code. Any suggestions, thoughts, workarounds?

**Comment by atomiks** [comment]
It would be an issue with `react-aria` as this library just returns regular props to be spread. 

I don't know how react-aria works but it doesn't seem like clicks or presses work at all, at least in the demo:
```js
    const { buttonProps } = useButton(
      {
        ...props,
        // neither of these are logged when using Enter/Space
        onClick() {
          console.log("clicked");
        },
        onPress() {
          console.log("pressed");
        },
      },
      internalRef,
    );
```

**Comment by nygardk** [comment]
Yeah, it seems that `onClick` passed for `useButton` only works on click but not on Space/Enter press. It's understandable, since `onClick` isn't supposed to be used `useButton`, but only `usePress`. (It has been elaborated on their [blog](https://react-spectrum.adobe.com/blog/building-a-button-part-1.html) _why_ this is the case.)

`useButton` does some trickery behind the scenes, stops propagation of events and fires handlers in custom ways. This is likely why `onClick` is not working as expected on key press even though it "works" on click. Then, semantically speaking, "on click" is not even same as "on key press", so I somewhat understand why it's not fired on key press; they decided not to fire `onClick` on `onKeyUp` (likely) as naturally the events are also different. 

This makes me think that maybe `floating-ui` is relying too much on `onClick` interaction. Would you entertain an idea of a hook to replace `useClick` and to work with `react-aria`? This hook could e.g. one of the following interfaces:
A) `useCustomSomething`: inject custom handlers to replace whatever is needed in `useClick` currently to create your own "adapter"
B) `usePress`: ready out of the box with no configuration to work with `onPress*` handlers implemented by `react-aria`
C) `useKeyPress` or similar to work in conjuction with `useClick`

WDYT?

**Comment by atomiks** [comment]
I'm open to any improvements if you have a potential implementation draft PR - converted this to a discussion
